src/storage/leveldb/wallet_index.ts

```diff
@@ -48,47 +48,29 @@ export default class LevelWalletIndex implements IKVWalletIndex {
    * Convert a number to a uint32 buffer before saving on database.
    * Internal helper method, since this logic is used in multiple places.
    *
-   * @param {'access'|'wallet'} dest Which database to use.
    * @param key The key to use when setting the value.
    * @param {number} value The value to set.
    */
-  async _setNumber(dest: 'access'|'wallet', key: string, value: number) {
+  async _setNumber(key: string, value: number) {
     if (value < 0) {
       throw new Error(`Invalid unsigned int ${value} being set on ${key}`);
     }
-    const buf = Buffer.alloc(4);
-    buf.writeUInt32BE(value);
-    switch(dest) {
-      case 'access':
-        await this.accessDB.put<string, Buffer>(key, buf, { valueEncoding: 'buffer'});
-        break;
-      case 'wallet':
-        await this.walletDB.put<string, Buffer>(key, buf, { valueEncoding: 'buffer'});
-        break;
-    }
+
+    const val = value.toString(16).padStart(8, '0');
+    await this.walletDB.put(key, val);
   }
 
   /**
    * Get the number from the uint32 buffer saved on database.
    * Internal helper method, since this logic is used in multiple places.
    *
-   * @param {'access'|'wallet'} dest Which database to fetch from.
    * @param {string} key The key to fetch.
    * @returns {Promise<number|null>}
    */
-  async _getNumber(dest: 'access'|'wallet', key: string): Promise<number|null> {
+  async _getNumber(key: string): Promise<number|null> {
     try {
-      let buf: Buffer;
-      switch(dest) {
-        case 'access':
-          buf = await this.accessDB.get<string, Buffer>(key, { valueEncoding: 'buffer'});
-          break;
-        case 'wallet':
-          buf = await this.walletDB.get<string, Buffer>(key, { valueEncoding: 'buffer'});
-          break;
-      }
-
-      return buf.readUint32BE(0);
+      const tmp = await this.walletDB.get(key);
+      return Number('0x'+tmp);
```

Comment:
Just to be extra sure on this implicit conversion results, I'd suggest using string templates:
```suggestion
      return Number(`$0x${tmp}`);
```

Reply:
Why the extra `$` at the start?